tests/MongoDB.Driver.Tests/Specifications/client-side-encryption/prose-tests/ClientEncryptionProseTests.cs

```diff
@@ -0,0 +1,833 @@
+            if (async)
+            {
+                encryptedValue = clientEncryption
+                    .EncryptAsync(
+                        value,
+                        encryptOptions,
+                        CancellationToken.None)
+                    .GetAwaiter()
+                    .GetResult();
+            }
+            else
+            {
+                encryptedValue = clientEncryption.Encrypt(
+                    value,
+                    encryptOptions,
+                    CancellationToken.None);
+            }
+
+            return encryptedValue;
+        }
+
+        private IAsyncCursor<BsonDocument> Find(
+            IMongoCollection<BsonDocument> collection,
+            BsonDocument filter,
+            bool async)
+        {
+            if (async)
+            {
+                return collection
+                    .FindAsync(new BsonDocumentFilterDefinition<BsonDocument>(filter))
+                    .GetAwaiter()
+                    .GetResult();
+            }
+            else
+            {
+                return collection
+                    .FindSync(new BsonDocumentFilterDefinition<BsonDocument>(filter));
+            }
+        }
+
+        private IMongoCollection<BsonDocument> GetCollection(IMongoClient client, CollectionNamespace collectionNamespace)
+        {
+            return client
+                .GetDatabase(collectionNamespace.DatabaseNamespace.DatabaseName)
+                .GetCollection<BsonDocument>(collectionNamespace.CollectionName);
+        }
+
+        private IReadOnlyDictionary<string, IReadOnlyDictionary<string, object>> GetKmsProviders()
+        {
+            var kmsProviders = new Dictionary<string, IReadOnlyDictionary<string, object>>();
+
+            var awsRegion = Environment.GetEnvironmentVariable("FLE_AWS_REGION") ?? "us-east-1";
+            var awsAccessKey = Environment.GetEnvironmentVariable("FLE_AWS_ACCESS_KEY_ID") ?? throw new Exception("The AWS_ACCESS_KEY_ID system variable should be configured on the machine.");
+            var awsSecretAccessKey = Environment.GetEnvironmentVariable("FLE_AWS_SECRET_ACCESS_KEY") ?? throw new Exception("The AWS_SECRET_ACCESS_KEY system variable should be configured on the machine.");
+            var kmsOptions = new Dictionary<string, object>
+            {
+                { "region", awsRegion },
+                { "accessKeyId", awsAccessKey },
+                { "secretAccessKey", awsSecretAccessKey }
+            };
+            kmsProviders.Add("aws", kmsOptions);
+
+            var localOptions = new Dictionary<string, object>
+            {
+                { "key", new BsonBinaryData(Convert.FromBase64String(LocalMasterKey)).Bytes }
+            };
+            kmsProviders.Add("local", localOptions);
+
+            return new ReadOnlyDictionary<string, IReadOnlyDictionary<string, object>>(kmsProviders);
+        }
+
+        private DisposableMongoClient GetMongoClient(
```

Comment:
Probably should be named:

    CreateMongoClient

since in all instances we are actually creating a new `MongoClient` and never "getting" an existing instance.

Reply:
done